src/regex_impl.hh

```diff
@@ -511,17 +512,18 @@ private:
             if (search and not m_found_match)
             {
                 if (start_desc and m_threads.next_is_empty())
-                    to_next_start(pos, config, *start_desc);
+                    to_next_start(pos, config, *start_desc, idle_func);
                 m_threads.push_next({first_inst, -1});
             }
             m_threads.swap_next();
         }
     }
 
-    static void to_next_start(Iterator& start, const ExecConfig& config, const StartDesc& start_desc)
+    static void to_next_start(Iterator& start, const ExecConfig& config, const StartDesc& start_desc, auto&& idle_func)
```

Comment:
Similarly, this should be fast enough even on very big buffer, checking for events on every byte is going to slow the general case quite a lot.